packages/common/src/hardhat/tasks/xchainSetup.ts

```diff
@@ -113,6 +113,47 @@ async function setupChildMessenger(
   }
 }
 
+async function setupOvmBasedL1Chain(hre_: any, chainName: string, chainId: number) {
```

Comment:
small suggestion: instead of passing `chainName` AND `chainId` here, just pass in `chainId` and then use the `switch(chainId)` logic and use a hardcoded mapping of chain ID to chain name. You could even refactor the switch statement in the `setup-l2-xchain` task and re-use it here!